libgraph/include/katana/GraphTopology.h

```diff
@@ -549,31 +552,41 @@ class KATANA_EXPORT ProjectedTopology : public GraphTopologyTypes {
     if (this == &projected_topo_) {
       return true;
     }
-    if (num_nodes() != projected_topo_.num_nodes()) {
+    if (NumNodes() != projected_topo_.NumNodes()) {
       return false;
     }
-    if (num_edges() != projected_topo_.num_edges()) {
+    if (NumEdges() != projected_topo_.NumEdges()) {
       return false;
     }
 
     return adj_indices_ == projected_topo_.adj_indices_ &&
            dests_ == projected_topo_.dests_;
   }
 
+  /// Gets all out-edges.
+  edges_range OutEdges() const noexcept {
+    return MakeStandardRange<edge_iterator>(Edge{0}, Edge{NumEdges()});
+  }
+
   /// Gets the edge range of some node.
   ///
   /// \param node node to get the edge range of
   /// \returns iterable edge range for node.
-  edges_range edges(Node node) const noexcept {
+  edges_range OutEdges(Node node) const noexcept {
     KATANA_LOG_DEBUG_ASSERT(node < adj_indices_.size());
     edge_iterator e_beg{node != 0 ? adj_indices_[node - 1] : 0};
     edge_iterator e_end{adj_indices_[node]};
 
     return MakeStandardRange(e_beg, e_end);
   }
 
-  Node edge_source(const Edge& eid) const noexcept {
-    KATANA_LOG_DEBUG_ASSERT(eid < num_edges());
+  Node OutEdgeDst(Edge edge_id) const noexcept {
+    KATANA_LOG_DEBUG_ASSERT(edge_id < dests_.size());
+    return dests_[edge_id];
+  }
+
+  Node GetEdgeSrc(const Edge& eid) const noexcept {
```

Comment:
In the RFC-114, we decided that `GetEdgeSrc` will have different semantics from `OutEdgeDst` and `InEdgeSrc`. It will accept a different index type (a local edge ID as opposed to a topology handle), and it is meant to support a different use case, where you don't walk over out-edges of some node, but rather over all edge IDs directly. So the user of this method won't know whether it is an out- or an in-edge. For now, the name may look a bit clumsy.